mutex.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package mutex
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/juju/errors"
+)
+
+var (
+	validName = regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9-]*$")
+)
+
+// Releaser defines the Release method that is the only thing that can be done
+// to a acquired mutex.
+type Releaser interface {
+	// Release makes the mutex available for others and invalidates the instance.
+	// Subsequent calls to Release will error.
+	Release() error
```

Comment:
Yeah, I don't think that one little utility package really gets to decide that the whole process is unviable just because (most likely scenario) someone deferred a second Release or something.

I know that many people are pretty goddamn lazy about `defer ThingThatErrors()`, and that does piss me off, but I don't think it's a very good reason to panic... I do think we should expect and anticipate pathologically stupid usage, and surely we're in a perfect position to DTRT by intercepting extra releases?